### PR TITLE
Suppress noisy "Loaded package environment ..." messages

### PR DESCRIPTION
**Pull request by mboes**
This is a workaround for
https://gitlab.haskell.org/ghc/ghc/merge_requests/1308.

### CODE REVIEW

File: haskell/toolchain.bzl [hunk 1]
```diff
@@ -80,7 +80,11 @@ while IFS= read -r line; do compile_flags+=("$line"); done < %s
 while IFS= read -r line; do extra_args+=("$line"); done < %s
 while IFS= read -r line; do param_file_args+=("$line"); done < %s
 
-"${compile_flags[@]}" "${extra_args[@]}" ${param_file_args+"${param_file_args[@]}"}
+# XXX Workaround https://gitlab.haskell.org/ghc/ghc/merge_requests/1308.
+output=$("${compile_flags[@]}" "${extra_args[@]}" ${param_file_args+"${param_file_args[@]}"} 2>&1)
+rc=$?
+echo "$output" | while IFS= read -r line; do [[ $line =~ ^Loaded ]] || echo "$line"; done
```
Comment (aherrmann) [inline haskell/toolchain.bzl]:
```suggestion
echo "$output" | while IFS= read -r line; do [[ $line =~ ^Loaded ]] || echo "$line" >&2; done
```
To keep `bazel build`'s stdout reasonably parseable.

File: haskell/toolchain.bzl [hunk 2]
```diff
@@ -80,7 +80,11 @@ while IFS= read -r line; do compile_flags+=("$line"); done < %s
 while IFS= read -r line; do extra_args+=("$line"); done < %s
 while IFS= read -r line; do param_file_args+=("$line"); done < %s
 
-"${compile_flags[@]}" "${extra_args[@]}" ${param_file_args+"${param_file_args[@]}"}
+# XXX Workaround https://gitlab.haskell.org/ghc/ghc/merge_requests/1308.
+output=$("${compile_flags[@]}" "${extra_args[@]}" ${param_file_args+"${param_file_args[@]}"} 2>&1)
```
Comment (aherrmann) [inline haskell/toolchain.bzl]: I think we can avoid storing the output in a variable and instead stream it right away.

```suggestion
set -o pipefail
"${compile_flags[@]}" "${extra_args[@]}" ${param_file_args+"${param_file_args[@]}"} 2>&1 \
  | while IFS= read -r line; do [[ $line =~ ^Loaded ]] || echo "$line"; done >&2
```

`set -o pipefail` forwards the exit code if GHC fails, so the `rc=$?` below becomes unnecessary.

**Comment by mboes** [comment]
Great suggestions, thanks.

File: haskell/toolchain.bzl [hunk 3]
```diff
@@ -80,7 +80,13 @@ while IFS= read -r line; do compile_flags+=("$line"); done < %s
...
 while IFS= read -r line; do param_file_args+=("$line"); done < %s
 
-"${compile_flags[@]}" "${extra_args[@]}" ${param_file_args+"${param_file_args[@]}"}
+# XXX Workaround https://gitlab.haskell.org/ghc/ghc/merge_requests/1308.
+set -o pipefail
+"${compile_flags[@]}" "${extra_args[@]}" ${param_file_args+"${param_file_args[@]}"} 2>&1 \
+  | while IFS= read -r line; do [[ $line =~ ^Loaded ]] || echo "$line"; done >&2
+rc=$?
+echo "$output" | while IFS= read -r line; do [[ $line =~ ^Loaded ]] || echo "$line" >&2; done
+exit $rc
```
Comment (aherrmann) [inline haskell/toolchain.bzl]: Those three lines then become redundant. (I don't know how to write multi-line suggestions on GH UI).

Comment (mboes) [reply haskell/toolchain.bzl]: Yeah I just noticed this was causing extraneous whitespace in the output. Fixed in the latest push.

**Pull request merged**